examples/java/core/serialization/src/main/java/com/microsoft/bond/examples/Serialization.java

```diff
@@ -0,0 +1,43 @@
+package com.microsoft.bond.examples;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+
+import com.microsoft.bond.Deserializer;
+import com.microsoft.bond.Serializer;
+import com.microsoft.bond.StructBondType;
+
+import com.microsoft.bond.protocol.CompactBinaryReader;
+import com.microsoft.bond.protocol.CompactBinaryWriter;
+
+import examples.serialization.Struct;
+
+public class Serialization {
+
+    public static void main(final String[] args) throws IOException {
+
+        final Struct obj = new Struct();
+        obj.n = 0x1000;
+        obj.str = "test";
+        obj.items = new ArrayList<>(2);
```

Comment:
`obj.items` should be an empty `ArrayList` by the time you get here, and will auto-resize as necessary. You don't need to do this, and it incorrectly implies that collection members aren't initialized by default.